opm/material/densead/EvaluationV.hpp

```diff
@@ -0,0 +1,602 @@
+// -*- mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*-
+// vi: set et ts=4 sw=4 sts=4:
+/*
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+
+  Consult the COPYING file in the top-level source directory of this
+  module for the precise wording of the license and the list of
+  copyright holders.
+*/
+/*!
+ * \file
+ *
+ * \brief Representation of an evaluation of a function and its derivatives w.r.t. a set
+ *        of variables in the localized OPM automatic differentiation (AD) framework.
+ */
+#ifndef OPM_DENSEAD_EVALUATION_V_HPP
+#define OPM_DENSEAD_EVALUATION_V_HPP
+
+#include "MathV.hpp"
+
+#include <opm/material/common/Valgrind.hpp>
+
+#include <dune/common/version.hh>
+
+#include <vector>
+#include <cmath>
+#include <cassert>
+#include <cstring>
+#include <iostream>
+#include <algorithm>
+
+namespace Opm {
+namespace DenseAd {
+
+/*!
+ * \brief Represents a function evaluation and its derivatives w.r.t. a fixed set of
+ *        variables.
+ */
+template <class ValueT>
+class EvaluationV
+{
+public:
+    //! field type
+    typedef ValueT ValueType;
+
+    //! number of derivatives
+    int size = 0;
```

Comment:
better make `size` a method, possibly renaming it to `numDerivatives()` because the current name is confusing IMO. (also for compile time evaluations.)

Reply:
Yes. It should have a better name.